frontend/src/common/navbar/index.tsx

```diff
@@ -1,111 +1,82 @@
-import * as React from 'react';
+import React, { useState } from 'react';
 import AppBar from '@mui/material/AppBar';
-import Box from '@mui/material/Box';
-import Divider from '@mui/material/Divider';
-import Drawer from '@mui/material/Drawer';
-import IconButton from '@mui/material/IconButton';
-import List from '@mui/material/List';
-import ListItem from '@mui/material/ListItem';
-import ListItemButton from '@mui/material/ListItemButton';
-import ListItemText from '@mui/material/ListItemText';
-import MenuIcon from '@mui/icons-material/Menu';
 import Toolbar from '@mui/material/Toolbar';
 import Typography from '@mui/material/Typography';
 import Button from '@mui/material/Button';
 import { useIsLoggedIn } from '@/hooks';
+import { AuthModal } from '@/components/common/AuthModal/AuthModal';
+import { useLogOut } from '@/hooks/useLogOut';
 
-interface NavbarProps {
-  navItems: string[];
-}
+const NavBar = () => {
+  const isLoggedIn = useIsLoggedIn();
+  const [performLogOut, logOutResult] = useLogOut();
 
-interface DrawerProps {
-  navItems: string[];
-  handleDrawerToggle: () => void;
-}
+  const [authModalOpen, setAuthModalOpen] = useState(false); // added this state
 
-const drawerWidth = 240;
-
-const DrawerElement = ({ handleDrawerToggle, navItems }: DrawerProps) => (
-  <Box onClick={handleDrawerToggle} sx={{ textAlign: 'center' }}>
-    <Typography variant="h6" sx={{ my: 2 }}>
-      MUI
-    </Typography>
-    <Divider />
-    <List>
-      {navItems.map((item: string) => (
-        <ListItem key={item} disablePadding>
-          <ListItemButton sx={{ textAlign: 'center' }}>
-            <ListItemText primary={item} />
-          </ListItemButton>
-        </ListItem>
-      ))}
-    </List>
-  </Box>
-);
-
-const NavBar = ({ navItems }: NavbarProps) => {
-  // const window = new Window();
-  const [mobileOpen, setMobileOpen] = React.useState(false);
-  const loggedIn = useIsLoggedIn();
-
-  console.log('logged in? ', loggedIn); //FOR DONGYI -> loggedIn == true -> user is logged in -> show profile, dashboard buttons etc.
-
-  const handleDrawerToggle = () => {
-    setMobileOpen((prevState) => !prevState);
+  const handleLogOut = async (event: React.MouseEvent) => {
+    event.preventDefault(); // prevent default anchor behavior
+    await performLogOut();
+    window.location.href = '/'; // redirect to home after logout
```

Comment:
Would prefer to use the `useNavigate` hook rather than the window variable. There are examples in other current PRs. It's a hook from the `react-router-dom` we are using to handle routing/navigation.

Reply:
Also, I reckon you wouldn't need the comments, your code is quite self documenting here.

Reply:
Will remove the comments, tried to use the Navigate but idk why it always be a blank page

Reply:
> Will remove the comments, tried to use the Navigate but idk why it always be a blank page

could you put a screenshot of what you were trying?

Reply:
The 19th line is used be like :
![image](https://github.com/mursal-ashraf/bankan/assets/100513480/7bb7bf68-3b09-4197-8d1d-4eb21f368631)

Reply:
can you also attach how you define `navigate`, thx.

Reply:
![image](https://github.com/mursal-ashraf/bankan/assets/100513480/ce1f73e1-dd38-4f98-8903-8d6a16ee734d)